src/meta/src/rpc/server.rs

```diff
@@ -171,316 +153,312 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
                 leader_addr.host,
                 leader_addr.port
             );
-
-            if note_status_leader_rx.changed().await.is_err() {
-                panic!("Issue receiving leader value from channel");
-            }
         }
     });
 
-    // FIXME: Start leader services if follower becomes leader
-    // failover logic
-
-    // Start follower services
-    if !leader_rx.borrow().1 {
-        tracing::info!("Node initially elected as follower");
-        tokio::spawn(async move {
-            let span = tracing::span!(tracing::Level::INFO, "services");
-            let _enter = span.enter();
-
-            let health_srv = HealthServiceImpl::new();
-            // run follower services until node becomes leader
-            tracing::info!("Starting follower services");
-            tokio::spawn(async move {
-                tonic::transport::Server::builder()
-                    .layer(MetricsMiddlewareLayer::new(Arc::new(MetaMetrics::new())))
-                    .add_service(HealthServer::new(health_srv))
-                    .serve(address_info.listen_addr)
-                    .await
-                    .unwrap();
-            });
-        });
-
-        let shutdown_election = async move {
-            if election_shutdown.send(()).is_err() {
-                tracing::warn!("election service already shut down");
-            } else if let Err(err) = election_handle.await {
-                tracing::warn!("Failed to join shutdown: {:?}", err);
-            }
+    // FIXME: maybe do not use a channel here
+    // What I need is basically a oneshot channel with one producer and multiple consumers
+    let (svc_shutdown_tx, svc_shutdown_rx) = WatchChannel(());
+
+    let join_handle = tokio::spawn(async move {
+        let span = tracing::span!(tracing::Level::INFO, "services");
+        let _enter = span.enter();
+
+        // failover logic
+        services_leader_rx
+            .changed()
+            .await
+            .expect("Leader sender dropped");
```

Comment:
I am keeping the `expect` here instead of returning. Let me know if your opinion differs.